src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/ElfObjectWriter.cs

```diff
@@ -414,6 +436,143 @@ private void EmitRelocationsARM64(int sectionIndex, List<SymbolicRelocation> rel
+            if (_machine == EM_ARM)
+            {
+                // Emit EABI attributes section
+                // (Addenda to, and Errata in, the ABI for the Arm Architecture, 2023Q3)
+                SectionWriter sectionWriter = GetOrCreateSection(ArmAttributesSection);
+                EabiAttributesBuilder attributesBuilder = new EabiAttributesBuilder(sectionWriter);
+                attributesBuilder.StartSection("aeabi");
+                attributesBuilder.WriteAttribute(Tag_conformance, "2.09");
+                attributesBuilder.WriteAttribute(Tag_CPU_name, "arm7tdmi");
+                attributesBuilder.WriteAttribute(Tag_CPU_arch, 2); // Arm v4T / Arm7TDMI
+                attributesBuilder.WriteAttribute(Tag_ARM_ISA_use, 1); // Yes
+                attributesBuilder.WriteAttribute(Tag_THUMB_ISA_use, 3); // Yes, determined by Tag_CPU_arch / Tag_CPU_arch_profile
+                attributesBuilder.WriteAttribute(Tag_ABI_PCS_R9_use, 0); // R9 is callee-saved register
+                attributesBuilder.WriteAttribute(Tag_ABI_PCS_RW_data, 1); // PC-relative
+                attributesBuilder.WriteAttribute(Tag_ABI_PCS_RO_data, 1); // PC-relative
+                attributesBuilder.WriteAttribute(Tag_ABI_PCS_GOT_use, 2); // indirect
+                attributesBuilder.WriteAttribute(Tag_ABI_FP_denormal, 1);
+                attributesBuilder.WriteAttribute(Tag_ABI_FP_exceptions, 0); // Unused
+                attributesBuilder.WriteAttribute(Tag_ABI_FP_number_model, 3); // IEEE 754
+                attributesBuilder.WriteAttribute(Tag_ABI_align_needed, 1); // 8-byte
+                attributesBuilder.WriteAttribute(Tag_ABI_align_preserved, 1); // 8-byte
+                attributesBuilder.WriteAttribute(Tag_CPU_unaligned_access, 0); // None
+                attributesBuilder.EndSection();
+            }
+        }
+
+        private protected override void CreateEhSections()
+        {
+            // ARM creates the EHABI sections lazily in EmitUnwindInfo
+            if (_machine is not EM_ARM)
+            {
+                base.CreateEhSections();
+            }
+        }
+
+        private protected override void EmitUnwindInfo(
+            SectionWriter sectionWriter,
+            INodeWithCodeInfo nodeWithCodeInfo,
+            string currentSymbolName)
+        {
+            if (_machine is not EM_ARM)
+            {
+                base.EmitUnwindInfo(sectionWriter, nodeWithCodeInfo, currentSymbolName);
+                return;
+            }
+
+            if (nodeWithCodeInfo.FrameInfos is FrameInfo[] frameInfos &&
+                nodeWithCodeInfo is ISymbolDefinitionNode)
+            {
+                bool shareSymbol = ShouldShareSymbol((ObjectNode)nodeWithCodeInfo);
+                SectionWriter exidxSectionWriter;
+                SectionWriter extabSectionWriter;
+
+                if (ShouldShareSymbol((ObjectNode)nodeWithCodeInfo) ||
+                    sectionWriter.SectionIndex != _managedCodeSectionIndex)
```

Comment:
What is this special casing of `_managedCodeSectionIndex` about?

Reply:
The specification allows either one `.ARM.exidx`/`.ARM.extab` section per code section, or one for each method. This is enforced to some extent by the format since the `.ARM.exidx` link field has to point to the corresponding code section.

Since we currently produce the unwinding info only for the managed code section this was merely a shortcut to use a single unwinding section for managed code and fallback to one per function for everything else (realistically only COMDAT handled by the first condition at this point).

Reply:
Alternatively I can update the code to properly produce `.ARM.exidx`/`.ARM.extab` for each section and get rid of the condition...

Reply:
> Alternatively I can update the code to properly produce .ARM.exidx/.ARM.extab for each section and get rid of the condition...

Done